design/host-discovery-controller/host-discovery.md

```diff
@@ -0,0 +1,325 @@
+state. The [state
+machine](https://github.com/metal3-io/baremetal-operator/blob/master/docs/baremetalhost-states.md)
+for the `baremetal-operator` will therefore need a new transition
+between `Discovered` and `MatchProfile` when the condition
+`!externallyProvisioned && !NeedsHardwareInspection()` is true. That
+transition will bypass the `Inspecting` state and allow the host to
+eventually move to the `Ready` state.
+
+### Implementation Details/Notes/Constraints
+
+The state management changes described above are likely to require
+changes to the `Provisioner` API. The details of those changes will be
+determined during implementation.
+
+The changes to the state machine would not change the fact that many
+of the metal3 features (fencing and power management) require BMC
+credentials. Discovered hosts would not enter the states where those
+features are allowed until BMC credentials are provided. The host
+controller should skip trying to maintain the power state and should
+not honor hard reboot requests for hosts without BMC credentials.
+
+### Risks and Mitigations
+
+If multiple `BareMetalHostDiscovery` resources exist, there may be
+multiple attempts to create a host from discovered information by
+different goroutines in the `host-discovery-controller`. If the name
+template information is the same, the conflict will be caught. If the
+name template information is different, then multiple hosts will be
+created. Resolving this will require adding a webhook in the
+`baremetal-operator` to be invoked when a resource is created and to
+block duplicate hosts. See [the v1alpha2
+migration](https://github.com/metal3-io/metal3-docs/pull/101) plans
+for details related to allowing webhooks. In the future, we may add
+rules to the `BareMetalHostDiscovery` API to control the naming
+convention applied to hosts with different characteristics, but there
+are no concrete plans to do so now.
+
+Data on discovered hosts will be lost if automatic provisioning is
+enabled. The feature is disabled by default, so the risk is mitigated
+because users will be aware when it is turned on.
+
+Ironic inspector creates discovered nodes with an `auto_discovered`
+flag set in the introspection data. We could limit the polling in the
+`baremetal-operator` to only look for hosts with the flag set. It
+isn’t really clear how useful that would be, so the description above
+does not assume such a limit.
+
+Nodes are removed from Ironic when the host CR is deleted. A host may
+be rediscovered if the data is deleted and the host is rebooted into
+the Ironic agent image. This could cause confusion, if the users
+intent was to remove the host completely. It could also be seen as a
+feature, if it is considered a way to reprovision a host.
+
+### Work Items
+
+1. Create the new git repository for the controller and API and set up
+   CI jobs.
+2. Write the new controller
+3. Update ironic-inspector-image to add unknown hosts using the `fake`
```

Comment:
Should be `manual-management` and we'll need to make a few updates to ironic-image to enable this mode (add `agent` power interface, add `noop` management interface, enable fast track and netboot fallback - see my demo linked above).